pkg/util/objectwatcher/objectwatcher.go

```diff
@@ -317,3 +324,20 @@ func (o *objectWatcherImpl) allowUpdate(clusterName string, desiredObj, clusterO
 
 	return false
 }
+
+// resourceDeepEqual checks if the desired object and the cluster object are equal.
+func (o *objectWatcherImpl) resourceDeepEqual(desiredObj, clusterObj *unstructured.Unstructured, clusterName string) (bool, error) {
```

Comment:
```suggestion
func (o *objectWatcherImpl) resourceDeepEqual(desiredObj, clusterObj *unstructured.Unstructured, clusterName string) (bool) {
```

I think this method should be non-disruptive, in other words, it defaults to returning false on any unexpected errors, let the program continue running.